pkg/store/github/github_test.go

```diff
@@ -4,8 +4,192 @@
 
 package github
 
-import "testing"
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-github/v43/github"
+	"github.com/matryer/is"
+	"github.com/migueleliasweb/go-github-mock/src/mock"
+	"github.com/nrkno/terraform-registry/pkg/core"
+)
 
 func TestGithubStore(t *testing.T) {
-	t.Skip("not sure how to test this")
+	t.Run("create GitHubStore", func(t *testing.T) {
+		is := is.New(t)
+		emptyResult := new(github.RepositoriesSearchResult)
+		total := 0
+		emptyResult.Total = &total
+		mockedHTTPClient := mock.NewMockedHTTPClient(
+			mock.WithRequestMatch(
+				mock.GetSearchRepositories,
+				emptyResult,
+			),
+		)
+
+		c := github.NewClient(mockedHTTPClient)
+		store := &GitHubStore{
+			ownerFilter: "test-owner",
+			topicFilter: "test-topic",
+			client:      c,
+			cache:       make(map[string][]*core.ModuleVersion),
+		}
+
+		err := store.ReloadCache(context.Background())
+		is.True(err == nil)
+	})
+
+	t.Run("create GitHubStore with github error", func(t *testing.T) {
+		is := is.New(t)
+		mockedHTTPClient := mock.NewMockedHTTPClient(
+			mock.WithRequestMatchHandler(
+				mock.GetSearchRepositories,
+				http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					mock.WriteError(
+						w,
+						http.StatusInternalServerError,
+						"github went belly up or something",
+					)
+				}),
+			),
+		)
+		c := github.NewClient(mockedHTTPClient)
+		store := &GitHubStore{
+			ownerFilter: "test-owner",
+			topicFilter: "test-topic",
+			client:      c,
+			cache:       make(map[string][]*core.ModuleVersion),
+		}
+		store.client = c
+		err := store.ReloadCache(context.Background())
+		is.True(err != nil)
```

Comment:
`is.NoErr(err)`

Reply:
it's actually testing that we have an error